src/components/Preloader.tsx

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+
+const Preloader = () => {
+  return (
+    <div className="fixed inset-0 z-50 flex items-center justify-center bg-black bg-opacity-50">
+      <div className="w-16 h-16 ease-linear border-8 border-t-8 border-gray-200 rounded-full loader"></div>
+      <style jsx>{`
+        .loader {
+          border-top-color: #3498db;
+          animation: spin 1s linear infinite;
+        }
+        @keyframes spin {
+          0% {
+            transform: rotate(0deg);
+          }
+          100% {
+            transform: rotate(360deg);
+          }
+        }
+      `}</style>
+    </div>
```

Comment:
To już chyba łatwiej wynieść od jakiegoś globalnego pliku css niż kisić to w JSX.